### PR TITLE
Fix: Remove Classy 

### PR DESCRIPTION
**Pull request by billypchan**
## What's new in this PR?

Remove Classy from the project.

## Notes

- [ ] Break down this PR in several parts for easier reviewing.

### CODE REVIEW

File: Wire-iOS/Resources/Classy/stylesheet.cas [hunk 1]
```diff
@@ -16,328 +16,3 @@
 *  along with this program. If not, see http://www.gnu.org/licenses/.
```
Comment (mikeger) [inline Wire-iOS/Resources/Classy/stylesheet.cas]: I guess the cas files can be removed at this point?

File: Wire-iOS/Sources/UserInterface/ContactsUI/ContactsSectionHeaderView+Style.swift [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+extension ContactsSectionHeaderView {
+@objc func setupStyle() {
```
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/ContactsUI/ContactsSectionHeaderView+Style.swift]: Indentation is off.

File: Wire-iOS/Sources/UserInterface/ContactsUI/ContactsSectionHeaderView+Style.swift [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    titleLabel.font = .smallSemiboldFont
+    titleLabel.textColor = .textForeground //$color-text-foreground-dark; ?
```
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/ContactsUI/ContactsSectionHeaderView+Style.swift]: Comments left.

File: Wire-iOS/Sources/UserInterface/Conversation/Content/Previews/LocationPreviewController.swift [hunk 4]
```diff
@@ -50,6 +51,10 @@ class LocationPreviewController: TintColorCorrectedViewController {
         fatalError("init(coder:) has not been implemented")
     }
 
+    deinit {
+        mapView = nil
```
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/Previews/LocationPreviewController.swift]: This is not necessary, since the default implementation is doing it with all properties anyway.

File: Wire-iOS/Sources/UserInterface/Location/LocationSelectionViewController.swift [hunk 5]
```diff
@@ -68,12 +68,21 @@ import CoreLocation
         fatalError("init(coder:) has not been implemented, user 'init(forPopoverPresentation:)'")
     }
 
+    deinit {
+        locationManager.stopUpdatingLocation()
+
+        mapView.delegate = nil
+        mapView = nil
```
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/Location/LocationSelectionViewController.swift]: I think that this line is not necessary, since the ARC is inserting it anyway.

File: WireExtensionComponents/Utilities/ColorScheme.h [hunk 6]
```diff
@@ -28,10 +28,10 @@ extern ColorSchemeColor ColorSchemeColorTextBackground;
 extern ColorSchemeColor ColorSchemeColorTextDimmed;
 extern ColorSchemeColor ColorSchemeColorTextPlaceholder;
 
-extern ColorSchemeColor ColorSchemeColorAccent;
-extern ColorSchemeColor ColorSchemeColorAccentDimmed;
-extern ColorSchemeColor ColorSchemeColorAccentDimmedFlat;
-extern ColorSchemeColor ColorSchemeColorAccentDarken;
+//extern ColorSchemeColor ColorSchemeColorAccent; ///TODO: private
```
Comment (mikeger) [inline WireExtensionComponents/Utilities/ColorScheme.h]: Commented code left.

File: WireExtensionComponents/Utilities/ColorScheme.m [hunk 7]
```diff
@@ -20,10 +20,10 @@
 #import "UIColor+Mixing.h"
 #import "UIColor+WAZExtensions.h"
 
-NSString * const ColorSchemeColorAccent = @"accent-current";
-NSString * const ColorSchemeColorAccentDimmed = @"accent-current-dimmed";
-NSString * const ColorSchemeColorAccentDimmedFlat = @"accent-current-dimmed-flat";
-NSString * const ColorSchemeColorAccentDarken = @"accent-current-darken";
+//NSString * const ColorSchemeColorAccent = @"accent-current";
```
Comment (mikeger) [inline WireExtensionComponents/Utilities/ColorScheme.m]: Commented code left.

File: WireExtensionComponents/Utilities/ColorScheme.m [hunk 8]
```diff
@@ -247,10 +247,10 @@ - (NSDictionary *)colorSchemeColorsWithAccentColor:(UIColor *)accentColor colorS
     UIColor *lightGraphiteAlpha64 = [UIColor wr_colorFromString:@"rgb(141, 152, 159, 0.64)"];
 
     NSMutableDictionary *lightColors = [NSMutableDictionary dictionaryWithDictionary:
-                                @{ ColorSchemeColorAccent: accentColor,
-                                   ColorSchemeColorAccentDimmed: [accentColor colorWithAlphaComponent:0.16],
-                                   ColorSchemeColorAccentDimmedFlat: [[accentColor colorWithAlphaComponent:0.16] removeAlphaByBlendingWithColor:white],
-                                   ColorSchemeColorAccentDarken: [[accentColor mix:[UIColor blackColor] amount:0.1] colorWithAlphaComponent:0.32],
+                                @{// ColorSchemeColorAccent: accentColor,
```
Comment (mikeger) [inline WireExtensionComponents/Utilities/ColorScheme.m]: Commented code.

File: WireExtensionComponents/Utilities/ColorScheme.m [hunk 9]
```diff
@@ -306,10 +306,10 @@ - (NSDictionary *)colorSchemeColorsWithAccentColor:(UIColor *)accentColor colorS
     }
     
     NSMutableDictionary *darkColors = [NSMutableDictionary dictionaryWithDictionary:
-                               @{ ColorSchemeColorAccent: accentColor,
-                                  ColorSchemeColorAccentDimmed: [accentColor colorWithAlphaComponent:0.16],
-                                  ColorSchemeColorAccentDimmedFlat: [[accentColor colorWithAlphaComponent:0.16] removeAlphaByBlendingWithColor:backgroundGraphite],
-                                  ColorSchemeColorAccentDarken: [[accentColor mix:[UIColor blackColor] amount:0.1] colorWithAlphaComponent:0.32],
+                               @{ //ColorSchemeColorAccent: accentColor,
```
Comment (mikeger) [inline WireExtensionComponents/Utilities/ColorScheme.m]: Commented code.

File: Wire-iOS/Sources/UserInterface/ContactsUI/ContactsSectionHeaderView+Style.swift (hunk 3 above)
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/ContactsUI/ContactsSectionHeaderView+Style.swift]: Comment is from the `stylehsheet.cas`? If so shouldn't we use the dark variant here like: `UIColor(scheme: .textForeground, variant: .dark)`

File: Wire-iOS/Sources/UserInterface/Conversation/Content/Previews/LocationPreviewController.swift (hunk 4 above)
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/Previews/LocationPreviewController.swift]: Why is this necessary?

File: Wire-iOS/Sources/UserInterface/Location/LocationSelectionViewController.swift (hunk 5 above)
Comment (billypchan) [reply Wire-iOS/Sources/UserInterface/Location/LocationSelectionViewController.swift]: These mapView = nil lines are some hacks try to fix the crash after running testThatItRendersCorrectlyShareViewController_LocationMessage. But they seem not fix the issue and I will undo all of them.

File: Wire-iOS/Sources/UserInterface/Helpers/UIColor+Accent.swift [hunk 10]
```diff
@@ -19,8 +19,21 @@
...
-        return self.accent().mix(.black, amount: 0.32)
+    @objc static var accentDarken: UIColor {
+        return accent().mix(.black, amount: 0.1).withAlphaComponent(0.32)
+    }
+
+    @objc static var accentDimmedFlat: UIColor {
+        return accent().withAlphaComponent(0.16).removeAlphaByBlending(with: .white)
+    }
+
+    @objc static func buttonEmptyText(variant: ColorSchemeVariant) -> UIColor {
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/Helpers/UIColor+Accent.swift]: This could be a `ColorSchemeColor`

File: Wire-iOS/Sources/UserInterface/Location/LocationSelectionViewController.swift [hunk 11]
```diff
@@ -68,12 +68,21 @@ import CoreLocation
         fatalError("init(coder:) has not been implemented, user 'init(forPopoverPresentation:)'")
     }
 
+    deinit {
+        locationManager.stopUpdatingLocation()
+
+        mapView.delegate = nil
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/Location/LocationSelectionViewController.swift]: Also here `delegate` on `MKMapView` is weak so this shouldn't be necessary.

File: WireExtensionComponents/Utilities/ColorScheme.h (hunk 6 above)
Comment (typfel) [inline WireExtensionComponents/Utilities/ColorScheme.h]: Delete?

File: WireExtensionComponents/Utilities/ColorScheme.h [hunk 12]
```diff
@@ -50,7 +50,7 @@ extern ColorSchemeColor ColorSchemeColorChatHeadTitleText;
 extern ColorSchemeColor ColorSchemeColorChatHeadSubtitleText;
 
 extern ColorSchemeColor ColorSchemeColorButtonHighlighted;
-extern ColorSchemeColor ColorSchemeColorButtonEmptyText;
+//extern ColorSchemeColor ColorSchemeColorButtonEmptyText;
```
Comment (typfel) [inline WireExtensionComponents/Utilities/ColorScheme.h]: Why couldn't this stay a ColorSchemeColor?

Comment (billypchan) [reply WireExtensionComponents/Utilities/ColorScheme.h]: ColorSchemeColorButtonEmptyText is equal to accent color. I replace all accent color constant with the new method UIColor.accent(), which gives a correct accent color after the user chnages it in setting.

File: Wire-iOS/Sources/UserInterface/Helpers/UIColor+Accent.swift (hunk 10 above)
Comment (billypchan) [reply Wire-iOS/Sources/UserInterface/Helpers/UIColor+Accent.swift]: ColorSchemeColorButtonEmptyText was equal to accent color. I replace all accent color constant with the new method UIColor.accent(), which gives a correct accent color after the user changes it in the setting.

File: WireExtensionComponents/Utilities/ColorScheme.h (hunk 12 above)
Comment (typfel) [reply WireExtensionComponents/Utilities/ColorScheme.h]: You can still use a `ColorSchemeColor` since the color scheme is re-generated when the user changes his/her accent color.

**Comment by typfel** [comment]
@billypchan the issue comes from the fact the defaultColorScheme is shared between accounts. Doing a ColorScheme.default.updateColors() after switching accounts should fix the issue.

File: Wire-iOS/Sources/UserInterface/ContactsUI/ContactsSectionHeaderView+Style.swift (hunk 3 above)
Comment (billypchan) [reply Wire-iOS/Sources/UserInterface/ContactsUI/ContactsSectionHeaderView+Style.swift]: Checked the original Classy file and it should be textForeground

**Comment by typfel** [comment]
@billypchan from my side we can merge this now and we can fix the ColorScheme account switching bug in a separate PR>
